ui/src/app/applications/components/application-resource-tree/application-resource-tree.tsx

```diff
@@ -327,6 +327,7 @@ export const describeNode = (node: ResourceTreeNode) => {
 
 function processPodGroup(targetPodGroup: ResourceTreeNode, child: ResourceTreeNode, props: ApplicationResourceTreeProps) {
     const statusByKey = new Map<string, models.ResourceStatus>();
+    props.app.status?.resources?.forEach(res => statusByKey.set(nodeKey(res), res));
```

Comment:
Is it possible to construct the map more expressively in a single line, rather than creating and then populating the dictionary?

Reply:
Hi @crenshaw-dev , I cleaned this up some more.  So this pod group is different than the pod group in the Pods view.  I have a part two issue, https://github.com/argoproj/argo-cd/issues/9338, to make the pod group a common component.  The pod group in the tree/network view is different currently because the animation is nicer, and the placement of the various statuses are consistent with the other tree nodes. (See videos in the original PR https://github.com/argoproj/argo-cd/pull/8996) 

The `resourceStatus` and status in question are actually not used here when I render the pod group. I used the 'existing' way to render the status, here:  https://github.com/argoproj/argo-cd/blob/1246e156a0d07a49f397ae0653385decaee86d04/ui/src/app/applications/components/application-resource-tree/application-resource-tree.tsx#L423

Compare it to the pods view, where it uses this `resourceStatus` property:

https://github.com/argoproj/argo-cd/blob/1246e156a0d07a49f397ae0653385decaee86d04/ui/src/app/applications/components/application-pod-view/pod-view.tsx#L309

and

https://github.com/argoproj/argo-cd/blob/1246e156a0d07a49f397ae0653385decaee86d04/ui/src/app/applications/components/application-pod-view/pod-view.tsx#L105

I've updated the PR to get rid of it.  We'll have to refactor this when we do: https://github.com/argoproj/argo-cd/issues/9338